core/frontend/src/main/scala/io/udash/properties/seq/DirectSeqPropertyImpl.scala

```diff
@@ -29,8 +29,10 @@ class DirectSeqPropertyImpl[A](val parent: ReadableProperty[_], override val id:
     valueChanged()
   }
 
-  override def set(t: Seq[A]): Unit =
-    replace(0, properties.size, t: _*)
+  override def set(t: Seq[A], force: Boolean = false): Unit =
+    if (force || t == null || properties.size != t.size || t != get) {
```

Comment:
The third check is kind of evaluated again in the last one, isn't it? IMHO premature optimization.